tensorboard/uploader/exporter_test.py

```diff
@@ -227,6 +229,214 @@ def outdir_files():
             },
         )
 
+    def test_e2e_success_case_with_blob_sequence_data(self):
+        """Covers exporting of finished and unfinished blob sequences
+
+        as well as rpc error during blob streaming.
+        """
+        mock_api_client = self._create_mock_api_client()
+
+        def stream_experiments(request, **kwargs):
+            del request  # unused
+            self.assertEqual(kwargs["metadata"], grpc_util.version_metadata())
+
+            response = export_service_pb2.StreamExperimentsResponse()
+            response.experiments.add(experiment_id="123")
+            yield response
+            response = export_service_pb2.StreamExperimentsResponse()
+            response.experiments.add(experiment_id="456")
+            yield response
+
+        def stream_experiment_data(request, **kwargs):
+            self.assertEqual(kwargs["metadata"], grpc_util.version_metadata())
+
+            tag = "__default_graph__"
+            for run in ("train", "test"):
+                response = export_service_pb2.StreamExperimentDataResponse()
+                response.run_name = run
+                response.tag_name = tag
+                display_name = "%s:%s" % (request.experiment_id, tag)
+                response.tag_metadata.CopyFrom(
+                    test_util.scalar_metadata(display_name)
+                )
+                for step in range(1):
+                    response.blob_sequences.steps.append(step)
+                    response.blob_sequences.wall_times.add(
+                        seconds=1571084520 + step, nanos=862939144
+                    )
+                    blob_sequence = blob_pb2.BlobSequence()
+                    if run == "train":
+                        # A finished blob sequence.
+                        blob = blob_pb2.Blob(
+                            blob_id="%s_blob" % run,
+                            state=blob_pb2.BlobState.BLOB_STATE_CURRENT,
+                        )
+                        blob_sequence.entries.append(
+                            blob_pb2.BlobSequenceEntry(blob=blob)
+                        )
+                    elif run == "test":
+                        # An unfinihsed blob.
+                        blob_sequence.entries.append(
+                            # `blob` unspecified: An unfinished blob.
+                            blob_pb2.BlobSequenceEntry()
```

Comment:
This isn’t an unfinished blob; it’s a hole. It’s good to test for this;
can we also have a test for an actual unfinished blob: i.e., a blob with
state `blob_pb2.BLOB_STATE_UNFINALIZED`?

Reply:
Done adding a test condition for `BLOB_STATE_UNFINALIZED`.